tornado/locale.py

```diff
@@ -198,13 +199,10 @@ def load_gettext_translations(directory: str, domain: str) -> None:
     global _supported_locales
     global _use_gettext
     _translations = {}
-    for lang in os.listdir(directory):
-        if lang.startswith("."):
-            continue  # skip .svn, etc
-        if os.path.isfile(os.path.join(directory, lang)):
-            continue
+
+    for filename in glob.glob(os.path.join(directory, '*', 'LC_MESSAGES', domain + '.mo')):
+        lang = filename.split('/LC_MESSAGES/', 1)[0].split(directory, 1)[1].strip('/')
```

Comment:
It might be a bit more robust like so:

```python
_, lang = os.path.split(os.path.dirname(os.path.dirname(filename)))
```

I'm thinking of windows paths using `\`, or somewhat badly chosen directories resulting in file paths like "/myapp/LC_MESSAGES/en_US/LC_MESSAGES/myapp.mo".

Reply:
Jeah, great suggestion. Thanks. Could also have been `os.path.basename(os.path.dirname(os.path.dirname(filename)))`

Reply:
ah, yes, `basename()` does what we want

Reply:
(either way seems fine to me, thanks)